### PR TITLE
Feature/upgrades and bug fixes

### PR DESCRIPTION
**Pull request by Aleksandras-2021**
- Changed logic of how cultist picks up bodies (OnTriggerEnter2D now)
- Added upgrade controller
- Shrine changes visuals upon upgrade( more visuals are needed though)
- other ignorable things.

### CODE REVIEW

File: Assets/Scripts/Buildings/Graveyard.cs [hunk 1]
```diff
@@ -20,8 +21,12 @@ private void OnTriggerEnter2D(Collider2D other)
     {
         if (other.tag == "DeadBody")
         {
+            DeadBody deadbody = other.GetComponent<DeadBody>();
+
             _shopManager.AddSouls(1);
-            other.GetComponent<DeadBody>().FreeToPool();
+            _deadBodyDeliveredEventSO.value = other.GetComponent<DeadBody>();
```
Comment (DzefersonS) [inline Assets/Scripts/Buildings/Graveyard.cs:27]:
```c#
_deadBodyDeliveredEventSO.value = deadbody
```

File: Assets/Scripts/Managers/CultistManager.cs [hunk 2]
```diff
@@ -23,31 +22,44 @@ public class CultistManager : MonoBehaviour
     [SerializeField] private int CultistSpawnPositionXMaximum;
     [SerializeField] private float CultistSpawnPositionY;
 
+    [SerializeField] public int deadbodycount;
```
Comment (DzefersonS) [inline Assets/Scripts/Managers/CultistManager.cs:25]: public is serialized, remove serializefield

File: Assets/Scripts/NPCs/States/FleeState.cs [hunk 3]
```diff
@@ -6,11 +6,16 @@ public class FleeState : CultistBaseState
 {
     private Transform _enemyTransform;
 
-    private float _fleeDuration = 1.5f; // Duration the cultist should stay in FleeState
+    private float _fleeDuration = 3f; // Duration the cultist should stay in FleeState
```
Comment (DzefersonS) [inline Assets/Scripts/NPCs/States/FleeState.cs:9]: [SerializeField] to edit in inspector

**Review by DzefersonS — CHANGES_REQUESTED** [review]
fix and merge

File: Assets/Scripts/Buildings/Graveyard.cs (hunk 1 above)
Comment (Aleksandras-2021) [reply Assets/Scripts/Buildings/Graveyard.cs:27]: oopsie missed this one when fixing shit

File: Assets/Scripts/NPCs/States/FleeState.cs (hunk 3 above)
Comment (Aleksandras-2021) [reply Assets/Scripts/NPCs/States/FleeState.cs:9]: Made an SO variable in cultist data SO for this instead of serialize field.

**Pull request merged**